src/Hydra/JsonSchema/SchemaFactory.php

```diff
@@ -74,6 +74,17 @@ public function buildSchema(string $className, string $format = 'jsonld', string
             $items = $schema['items'];
             unset($schema['items']);
 
+            switch ($schema->getVersion()) {
+                case Schema::VERSION_JSON_SCHEMA:
+                    $mappingPropDefinition = ['type' => ['string', 'null']];
+                    break;
+                case Schema::VERSION_OPENAPI:
+                    $mappingPropDefinition = ['type' => 'string', 'nullable' => true];
+                    break;
+                default:
+                    $mappingPropDefinition = ['type' => 'string'];
```

Comment:
Could you initialize the property before the switch and rename it `$nullableStringDefinition`? In case we want to reuse it elsewhere.

Reply:
Fixed!